static_frame/core/index_hierarchy.py

```diff
@@ -1603,184 +2305,209 @@ def loc_searchsorted(self,
-            levels_src = self._levels
+        Return an IndexHierarchy with a new root (outer) level added.
+        '''
+        if self._recache:
+            self._update_array_cache()
+
+        index_cls = self._INDEX_CONSTRUCTOR if index_constructor is None else index_constructor._MUTABLE_CONSTRUCTOR # type: ignore
+
+        if self.STATIC:
+            indices = [index_cls((level,)), *self._indices]
         else:
-            levels_src = self._levels.to_index_level()
+            indices = [index_cls((level,)), *(idx.copy() for idx in self._indices)]
 
-        index_cls = self._INDEX_CONSTRUCTOR if index_constructor is None else index_constructor
+        indexers = np.array(
+                [
+                    np.zeros(self.__len__(), dtype=DTYPE_INT_DEFAULT),
+                    *self._indexers
+                ]
+        )
+        indexers.flags.writeable = False
 
-        levels = self._LEVEL_CONSTRUCTOR(
-                index=index_cls((level,)),
-                targets=ArrayGO([levels_src], own_iterable=True),
-                offset=0,
-                own_index=True
-                )
-        # can transfrom TypeBlocks appropriately and pass to constructor
-        if not self._recache: # if we have TypeBlocks
-            array = np.full(self.__len__(), level)
-            array.flags.writeable = False
-            blocks = TypeBlocks.from_blocks(chain((array,), self._blocks._blocks))
-            return self.__class__(levels,
-                    name=self._name,
-                    blocks=blocks,
-                    own_blocks=True)
+        def gen_blocks() -> tp.Iterator[np.ndarray]:
+            # First index only has one value. Extract from array (instead of using `level`) since the constructor might have modified its type
+            yield np.full(self.__len__(), indices[0][0])
+            yield from self._blocks._blocks
 
-        return self.__class__(levels, name=self._name)
+        return self.__class__(
+                indices=indices,
+                indexers=indexers,
+                name=self.name,
+                blocks=TypeBlocks.from_blocks(gen_blocks()),
+                own_blocks=True,
+                )
 
-    def level_drop(self,
+    def level_drop(self: IH,
             count: int = 1,
-            ) -> tp.Union[Index, 'IndexHierarchy']:
-        '''Return an IndexHierarchy with one or more leaf levels removed. This might change the size of the resulting index if the resulting levels are not unique.
+            ) -> tp.Union[Index, IH]:
+        '''
+        Return an IndexHierarchy with one or more leaf levels removed.
 
         Args:
             count: A positive value is the number of depths to remove from the root (outer) side of the hierarchy; a negative value is the number of depths to remove from the leaf (inner) side of the hierarchy.
         '''
         if self._recache:
-            self._update_array_cache()
+            self._update_array_cache() # pragma: no cover
```

Comment:
Why no coverage?

Reply:
See this response: https://github.com/InvestmentSystems/static-frame/pull/437#discussion_r831501537